### PR TITLE
chore: Add NSP AirGap

### PR DESCRIPTION
**Pull request by odinnordico**

### CODE REVIEW

File: ns-provisioner-samples/gitops-airgap/install/namespaces.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#! NOTE: if you have another tool like Tanzu Mission Control or some other process that is taking care of creating namespaces for you, 
+#! and you don’t want namespace provisioner to create the namespaces, you can delete this file from your GitOps install repository.
+#@ for ns in data.values.namespaces:
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: #@ ns.name
+  annotations:
+    kapp.k14s.io/create-strategy: fallback-on-update
```
Comment (atmandhol) [inline ns-provisioner-samples/gitops-airgap/install/namespaces.yaml:12]: What are these kapp annotations? do we need this?

File: ns-provisioner-samples/gitops-airgap/resources/settings.xml.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  provider: sample
+  #@yaml/text-templated-strings
+  settings.xml: |
+    <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
+        <mirrors>
+            <mirror>
+                <id>reposilite</id>
+                <name>Tanzu seal Internal Repo</name>
+                <url>https://reposilite.tap-trust.cf-app.com/releases</url>
```
Comment (atmandhol) [inline ns-provisioner-samples/gitops-airgap/resources/settings.xml.yaml:19]: Should this be templated to host?

File: ns-provisioner-samples/gitops-airgap/resources/tekton-pipeline-java.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - name: source-revision
+  tasks:
+    - name: test
+      params:
+        - name: source-url
+          value: $(params.source-url)
+        - name: source-revision
+          value: $(params.source-revision)
+      taskSpec:
+        params:
```
Comment (atmandhol) [inline ns-provisioner-samples/gitops-airgap/resources/tekton-pipeline-java.yaml:28]: We should add the securityContext change from this MR in this template as well https://github.com/vmware-tanzu/application-accelerator-samples/pull/377/files

File: ns-provisioner-samples/gitops-airgap/install/namespaces.yaml (hunk 1 above)
Comment (odinnordico) [reply ns-provisioner-samples/gitops-airgap/install/namespaces.yaml:12]: If the namespaces is manually created we own it

File: ns-provisioner-samples/gitops-airgap/resources/tekton-pipeline-java.yaml (hunk 3 above)
Comment (odinnordico) [reply ns-provisioner-samples/gitops-airgap/resources/tekton-pipeline-java.yaml:28]: Added

File: ns-provisioner-samples/gitops-airgap/resources/settings.xml.yaml (hunk 2 above)
Comment (odinnordico) [reply ns-provisioner-samples/gitops-airgap/resources/settings.xml.yaml]: Not sure how settings.xml works, added as a template value

File: ns-provisioner-samples/gitops-airgap/resources/settings.xml.yaml
Comment (shaheerkootteeri) [inline ns-provisioner-samples/gitops-airgap/resources/settings.xml.yaml:1]: should the file name be changed to ns-provisioner-samples/gitops-airgap/resources/settings-xml.yaml

Comment (odinnordico) [reply ns-provisioner-samples/gitops-airgap/resources/settings.xml.yaml:1]: There is no need to change it, but changed

**Review by shaheerkootteeri — APPROVED** [review]
LGTM

File: ns-provisioner-samples/gitops-airgap/resources/settings-xml.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  type: maven
+  provider: sample
+  #@yaml/text-templated-strings
+  settings.xml: |
+    <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
+        <mirrors>
+            <mirror>
+                <id>reposilite</id>
+                <name>Tanzu seal Internal Repo</name>
```
Comment (atmandhol) [inline ns-provisioner-samples/gitops-airgap/resources/settings-xml.yaml:18]: Should we make this name something more generic?

Comment (odinnordico) [reply ns-provisioner-samples/gitops-airgap/resources/settings-xml.yaml]: Changed to `Accelerator samples`

**Review by atmandhol — APPROVED** [review]

**Pull request merged**